CRM/Utils/System/WordPress.php

```diff
@@ -783,15 +782,22 @@ public function createUser(&$params, $mail) {
       }
     }
 
+    $this->hooks_core_remove();
     $uid = wp_insert_user($user_data);
 
     $creds = [];
     $creds['user_login'] = $params['cms_name'];
-    $creds['user_password'] = $params['cms_pass'];
     $creds['remember'] = TRUE;
-    $user = wp_signon($creds, FALSE);
 
-    wp_new_user_notification($uid, $user_data['user_pass']);
+    // Call wp_signon if we aren't already logged in
+    // For example, we might be creating a new user from the Contact record.
+    if (!current_user_can('create_users')) {
+      wp_signon($creds, FALSE);
+    }
+
+    wp_new_user_notification($uid, NULL, 'both');
```

Comment:
@mlutfy FWIW I agree with this proposal. Using `do_action('register_new_user')` instead of calling `wp_new_user_notification()` directly seems like the correct way of doing this.